server/controllers/reviewController/reviewRepository.js

```diff
@@ -0,0 +1,45 @@
+const ReviewDataRepository = require('../reviewDataConroller/reviewDataRepository');
+const { ReviewData } = require('../../db/models/Review');
+
+class ReviewRepository {
+    constructor(workspaceId) {
+        this.workspaceId = workspaceId != null ? workspaceId : null;
+        this.reviewDataRepository = new ReviewDataRepository(this.workspaceId);
+    }
+
+    async get() {
+        if (this.workspaceId != null) {
+            const reviewData = await ReviewData.findOne({ workspaceId: this.workspaceId });
+            return reviewData.reviews;
+        }
+    }
+
+    async create(review) {
+        if (this.workspaceId != null) {
+            const reviewData = await ReviewData.findOne({ workspaceId: this.workspaceId });
+            reviewData.reviews.push(review);
+
+            const updatedReviewData = await reviewData.save();
+            return updatedReviewData.reviews[updatedReviewData.reviews.length - 1];
+        }
+    }
+
+    async update(updatedReview) {
+        const reviewData = await this.reviewDataRepository.getById();
+        const reviewIndex = reviewData.reviews.findIndex((review) => String(review._id) === String(updatedReview._id));
+
+        reviewData.reviews[reviewIndex] = updatedReview;
+        const updatedReviewData = await reviewData.save();
+        return updatedReviewData.reviews[reviewIndex];
+    }
+
+    async delete(reviewId) {
+        const reviewData = await this.reviewDataRepository.getById();
+        const updatedReviews = reviewData.reviews.filter((review) => String(reviewId) !== String(review._id));
```

Comment:
Maybe getById() could take an ID as input and then you wouldn't have the .filter step?

Reply:
This will be a change I make when I change databases, as the structure the current database does not allow me to query a single review by its id, I have to find the ReviewData object, and find the review by iterating through the "reviews" property which is an array of reviews. This is why I personally prefer working with SQL databases, much easier to manage relational logic.